src/components/admin.tsx

```diff
@@ -759,21 +762,25 @@ function Admin({ user }: adminProps) {
     hours = DEFAULT_SELF_DESTRUCT_HOURS
   ) => {
     if (navigator.share) {
-      setIsSettingAssignLink(true);
       try {
         await navigator.share({
           title: title,
           text: body,
           url: url
         });
+        setSelectedPostal(postalcode);
+        if (linktype === LINK_TYPES.ASSIGNMENT) setIsSettingAssignLink(true);
```

Comment:
There is an issue with the loading indicator. Clicking personal link activates the loader in the assign button.